models/classes/Security/Business/Domain/Exception/SecretKeyGenerationException.php

```diff
@@ -0,0 +1,32 @@
+<?php declare(strict_types=1);
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
+namespace oat\taoLti\models\classes\Security\Business\Domain\Exception;
+
+use Exception;
+use RuntimeException;
+
+final class SecretKeyGenerationException extends RuntimeException
+{
+    public static function create(Exception $previous): self
```

Comment:
why new static method instaed of redeclaring __construct?

Reply:
https://github.com/oat-sa/tao-core/pull/2444/files#r396250943 :)

Reply:
even why do you need this method at all if you take everything from `parent`?

Reply:
Just to take everything from parent. The method is a simple factory.